localstack/services/stepfunctions/asl/component/state/state_execution/state_map/iteration/job.py

```diff
@@ -59,14 +68,14 @@ def next_job(self) -> Optional[Any]:
     def _is_terminated(self) -> bool:
         return len(self._closed_jobs) == self._jobs_number or self._worker_exception is not None
 
-    def _notify_on_termination(self):
+    def _notify_on_termination(self) -> None:
         if self._is_terminated():
             self._termination_event.set()
 
     def get_worker_exception(self) -> Optional[Exception]:
         return self._worker_exception
 
-    def close_job(self, job: Job):
+    def close_job(self, job: Job) -> None:
         with self._mutex:
```

Comment:
sidenote: I'm not familiar enough with the context to judge whether locking works correctly in all circumstances, but the main issue here seemed related to the serialization of a closed job (i.e., finished job), rather than logic-locking.